octobot_trading/data_manager/orders_manager.py

```diff
@@ -56,13 +56,14 @@ def upsert_order(self, order_id, raw_order) -> (bool, bool):
             return True, False
         return self._update_order_from_raw(self.orders[order_id], raw_order), True
 
-    def upsert_order_close(self, order_id, raw_order) -> bool:
+    def upsert_order_close(self, order_id, raw_order) -> (bool, Order):
         if self.has_order(order_id):
+            order = self.orders[order_id]
             self._update_order_from_raw(self.orders[order_id], raw_order)
-            # TODO order -> trade
             self.orders.pop(order_id)
-            return True
-        return False
+            # trade creation should be handled right after this
+            return True, order
```

Comment:
It can only be `True` if the order existed so I think we can simplify by returning only `order` or `None` and changed will be equals to `order is not None`.